src/app-layer-register.c

```diff
@@ -189,6 +189,13 @@ int AppLayerRegisterParser(const struct AppLayerParser *p, AppProto alproto)
     return 0;
 }
 
+int AppLayerRegisterParserAlias(const char *proto_name, const char *proto_alias)
+{
+    AppLayerProtoDetectRegisterAlias(proto_name, proto_alias);
```

Comment:
Why does it make sense to have 2 functions ? `AppLayerRegisterParserAlias ` and `AppLayerProtoDetectRegisterAlias `

Reply:
Thought that registering the aliases is semantically very similar to registering the protocols. Just thought that it's "the way" after seeing how AppLayerRegisterProtocolDetection() and AppLayerProtoDetectRegisterProtocol() are doing it.

That's why i placed alpda_ctx next to alpd_ctx in app-layer-detect-proto.c.

Reply:
ok thanks